src/cool_seq_tool/mappers/exon_genomic_coords.py

```diff
@@ -796,37 +801,14 @@ async def _genomic_to_tx_segment(
                 mane_transcripts = self.mane_transcript_mappings.get_gene_mane_data(
                     gene
                 )
-
-                if mane_transcripts:
+                if mane_transcripts and await self.uta_db.validate_mane_transcript_ac(
+                    mane_transcripts
+                ):
                     transcript = mane_transcripts[0]["RefSeq_nuc"]
                 else:
-                    # Attempt to find a coding transcript if a MANE transcript
-                    # cannot be found
-                    results = await self.uta_db.get_transcripts(
-                        gene=gene, alt_ac=genomic_ac
+                    transcript = await self._select_optimal_transcript(
```

Comment:
Few more thoughts:
Similar to the `ManeTranscript` class, I would consider adding `try_longest_compatible` argument. There might be cases where a user only wants MANE data to be returned. Since we're now mixing MANE/Longest compatible remaining, it might be good to include `TranscriptPriority` in the response, so users know if the transcript is a MANE transcript. Documentation should also be updated (in public method too)